.gitattributes

```diff
@@ -0,0 +1 @@
+﻿*.d.ts text
```

Comment:
The clue was here:

http://stackoverflow.com/questions/6855712/why-does-git-treat-this-text-file-as-a-binary-file

If git thinks a file is a text file, it will convert the line ends to \n when storing, and convert back to the operating system default when extracting. Under Windows, the file gets created with \r\n line ends by the compiler. git doesn't know it's text so it stores those \r\n untranslated in the repository - hence every line is a change.

the git attirbutes entry tell git that *.d.ts is in fact a text file, so on storing, it translates the line ends back to \n like it should.